File: app/src/main/java/dev/ragnarok/fenrir/model/AbsModel.java
package dev.ragnarok.fenrir.model;

import android.os.Parcel;
import android.os.Parcelable;

import androidx.annotation.CallSuper;


public abstract class AbsModel implements Parcelable {

    public AbsModel() {

    }

    @SuppressWarnings("unused")
    public AbsModel(Parcel in) {

    }

    @CallSuper
    @Override
    public void writeToParcel(Parcel parcel, int i) {

    }
}

File: app/src/main/kotlin/dev/ragnarok/fenrir/model/AbsModel.kt
package dev.ragnarok.fenrir.model

import android.os.Parcel
import android.os.Parcelable
import androidx.annotation.CallSuper

abstract class AbsModel : Parcelable {
    constructor()

    @Suppress("UNUSED_PARAMETER")
    constructor(`in`: Parcel?)

    @CallSuper
    override fun writeToParcel(parcel: Parcel, i: Int) {
    }
}